external-crates/move/crates/move-compiler/src/command_line/compiler.rs

```diff
@@ -96,21 +114,44 @@ enum PassResult {
     Typing(typing::ast::Program),
     HLIR(hlir::ast::Program),
     CFGIR(cfgir::ast::Program),
-    Compilation(Vec<AnnotatedCompiledUnit>, /* warnings */ Diagnostics),
+    Compilation(
+        Vec<AnnotatedCompiledUnit>,
+        Vec<ModuleIdent>,
+        /* warnings */ Diagnostics,
+    ),
 }
 
+pub type CompiledModuleInfoMap = BTreeMap<ModuleIdent, Arc<CompiledModuleInfo>>;
+
 #[derive(Clone)]
-pub struct FullyCompiledProgram {
-    pub files: MappedFiles,
-    pub parser: parser::ast::Program,
-    pub expansion: expansion::ast::Program,
-    pub naming: naming::ast::Program,
-    pub typing: typing::ast::Program,
-    pub hlir: hlir::ast::Program,
-    pub cfgir: cfgir::ast::Program,
-    pub compiled: Vec<AnnotatedCompiledUnit>,
+pub struct CompiledModuleInfo {
+    pub file_name: Symbol,
+    pub file_content: Arc<str>,
+    /// to compute address conflicts in expansion/translate.rs
+    /// (address map "inlined" into the module so that
+    /// modules are separated)
+    pub named_address_map: NamedAddressMap,
+    /// to create context in expansion/translate.rs
+    pub member_kinds: BTreeMap<Name, ModuleMemberKind>,
+    /// to build context in naming/translate.rs
+    pub resolved_members: BTreeMap<Symbol, ResolvedModuleMember>,
```

Comment:
Why do you need this? Isn't this information in `TypingInfo`?

Reply:
I think we can re-build this based on what's in `TypingInfo` - wasn't sure if we should lean more towards having things readily available in pre-compiled info or more towards keeping less data and spend more time recomputing things. After writing the code, though, don't think it matters too much as this is seemingly the only place and it should be fine to reconstruct it from `TypingInfo` at compilation time